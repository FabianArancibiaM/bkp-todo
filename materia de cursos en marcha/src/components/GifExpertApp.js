import React, { useState } from 'react';
import { AddCategory } from './category/AddCategory';
import { GifGrid } from './gif/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategory] = useState(['one pice']);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategory}/>
            <hr />
            
            <ol>
                {
                    categories.map( (category, index) =>
                        <GifGrid 
                        key={category}
                        category={category}
                        />
                    )
                }
            </ol>
        </>
    );
}

export default GifExpertApp;

