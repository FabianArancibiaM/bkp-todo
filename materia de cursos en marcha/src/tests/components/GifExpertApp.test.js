import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp'; 
describe('GifExpertApp Test Funciones', () => {
    let wraper = shallow(<GifExpertApp />);
    beforeEach( () => {
        wraper = shallow(<GifExpertApp />);
    });
    test('GifExpertApp', () => {
        expect(wraper).toMatchSnapshot();
    })
})