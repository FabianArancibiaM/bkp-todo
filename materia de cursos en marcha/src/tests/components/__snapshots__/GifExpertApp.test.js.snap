// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Test Funciones GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one pice"
      key="one pice"
    />
  </ol>
</Fragment>
`;
